lib/cancan/model_adapters/active_record_4_adapter.rb

```diff
@@ -6,6 +6,37 @@ def self.for_class?(model_class)
         model_class <= ActiveRecord::Base
       end
 
+      def self.override_association_matching?(subject, conditions)
+        subject.kind_of?(ActiveRecord::Relation) && !subject.loaded?
+      end
+
+      def self.matches_association?(subject, conditions)
+        where_conditions = association_conditions_for(subject.proxy_association.reflection, conditions)
+        subject.where(where_conditions).any?
+      end
+
+      def self.association_conditions_for(reflection, conditions)
+        {}.tap do |result|
+          target = reflection.klass
+          conditions.each_pair do |key, value|
+            key, value = association_condition_for(target, key, value)
+            result[key] = value
+          end
+        end
+      end
+
+      def self.association_condition_for(target, key, value)
+        string_key = key.to_s
+        if ActiveRecord::VERSION::MINOR >= 1 && target.defined_enums.include?(string_key)
```

Comment:
@Senjai Potentially in a future version maybe we should have one class each for AR4/AR4.1/AR4.2/AR5? AR5 won't need the enum hack, for one...